src/roboptim-core-plugin-knitro.cc

```diff
@@ -25,76 +24,14 @@
 
 #include "roboptim-core-plugin-knitro.hh"
 
+using namespace std;
+
 namespace roboptim
 {
 
-  template <typename U>
-  class EvaluateConstraint : public boost::static_visitor<>
-  {
-  public:
-    EvaluateConstraint (double* const c, std::ptrdiff_t& offset, const U& x)
-      : boost::static_visitor<> (),
-	c (c),
-	offset (offset),
-	x (x)
-    {}
-
-    template<typename T>
-    void
-    operator () (const boost::shared_ptr<T>& constraint) const
-    {
-      Eigen::Map<Eigen::Matrix<typename T::value_type,
-			       Eigen::Dynamic, 1> >
-	constraintValue (c + offset, constraint->outputSize ());
-    constraintValue = (*constraint) (x);
-
-    offset += constraint->outputSize ();
-    }
-
-  private:
-    double* const c;
-    std::ptrdiff_t& offset;
-    const U& x;
-  };
-
-  template <typename U>
-  class EvaluateConstraintJacobian : public boost::static_visitor<>
-  {
-  public:
-    EvaluateConstraintJacobian (int n,
-				int m,
-				double* const jac,
-				std::ptrdiff_t& offset,
-				const U& x)
-      : boost::static_visitor<> (),
-	n (n),
-	m (m),
-	jac (jac),
-	offset (offset),
-	x (x)
-    {}
-
-    template<typename T>
-    void
-    operator () (const boost::shared_ptr<T>& constraint) const
-    {
-      Eigen::Map<KNITROSolver::matrix_t> jac_ (jac, m, n);
-      jac_.block (offset, 0, constraint->outputSize (), n) =
-	constraint->jacobian (x);
-      offset += constraint->outputSize ();
-    }
-
-  private:
-    int n;
-    int m;
-    double* const jac;
-    std::ptrdiff_t& offset;
-    const U& x;
-  };
-
   static int KNITRO_callback (const int evalRequestCode,
 			      const int n,
-			      const int ROBOPTIM_DEBUG_ONLY (m),
+			      const int (m),
```

Comment:
The parentheses were used for the macro. If `m` is used in both debug and release, you can just use `const int m,`.